### DISCUSSION TITLE
Modify updatespeed room presence

### DISCUSSION THREAD

**Discussion by frankiej911** [mKeRix/room-assistant#634] [Q&A]
Hello,

Is it possible to modify the speed of the room presence detection? I have three Raspberry Pi's devided through my house, 1 in the livingroom (set-up with no maxDistance, this one also covers my kitchen), and 2 upstairs (the bedroom and the Office with a small maxDistance). After a lot of testing with the verbose parameter, adjusting with the maxDistance, the pysical locations of the Pi's and finetuning the measuredPower I've created a fairly reliable set-up and it detects the location of the iBeacon pretty well. The problem now is the slow detection in some rooms, especialy when I go upstairs. Is it possible to let Room Assistant switch faster if a Raspberry Pi has detected my iBeacon with a much stronger signal compared to the previous location? For instance in my Office the livingroom detects the iBeacon with a distance of 14. It switched after 2-3 minutes to my Office with a stable distance of max 2 meter... 

This is my configuration (iBeacons broadcasted measuredPower -59 and every 2 seconds, but testing with some overrides... The Pi on the groundfloor is a RPi2 with a USB Bluetooth adapter, the Pi's upstairs are 2 Zero W's) :

**Livingroom/Kitchen (ground floor):** 
cluster:
  port: 6425
  autoDiscovery: false
  quorum: 3
  peerAddresses:
    - 10.0.0.6:6425
    - 10.0.0.8:6425
  weight: 2

bluetoothLowEnergy:
  allowlist:
    - 11111111111111111111111111111111-1-1
  onlyIBeacon: true
(maxDistance: 9 disabled, little bit difficult with the kitchen)

**Bedroom**
cluster:
  port: 6425
  autoDiscovery: false
  quorum: 3
  peerAddresses:
    - 10.0.0.7:6425
    - 10.0.0.8:6425
  weight: 1

bluetoothLowEnergy:
  allowlist:
    - 11111111111111111111111111111111-1-1
  onlyIBeacon: true
  maxDistance: 2
  tagOverrides:
    11111111111111111111111111111111-1-1:
      measuredPower: -61

**Office**
cluster:
  port: 6425
  autoDiscovery: false
  quorum: 3
  peerAddresses:
    - 10.0.0.6:6425
    - 10.0.0.7:6425
  weight: 1

bluetoothLowEnergy:
  allowlist:
    - 11111111111111111111111111111111-1-1
  onlyIBeacon: true
  maxDistance: 2
  tagOverrides:
    11111111111111111111111111111111-1-1:
      measuredPower: -61

Thanks

**Comment by mKeRix** [comment]
Are you running on room-assistant v2.15.1? If so I'd be interested if you see the same behavior when you downgrade to the previous version (`sudo npm i --global --unsafe-perm room-assistant@2.15.0`). I just recently re-tuned the signal smoothing, which may affect detection speed (even though it shouldn't be this drastic!).

**Reply by frankiej911** [reply]
Hello, thanks. :) 

I will test the previous version this evening. It's not always 2-3 minutes for detection, but 20 seconds to 1 minute is a better average speed I think.  Maybe the MaxDistance also impacts the detection speed? I played a lot with this variables. It looks like (I'm not really sure, but will investigate it further) that the iBeacon goes to another room if the previous room is out of range from the MaxDistances point of view. The MaxDistance is disabled in the livingroom right now, maybe the 2-3 minutes was counted when I played with this values... The distance in Home Assistant goes slowly up or down (to reduce false positives I think), like you said about the signal smoothing. 

I'll keep you informed, thanks!
